File: backend/brackets/bracket_engine.py
"""
Public bracket engine API.

Implementation is split into smaller modules for maintainability; this file
re-exports the stable functions used by the Django API and unit tests.
"""

from __future__ import annotations

from .constants import MAX_BRACKET_INT_EXCLUSIVE
from .bracket_decode import bracket_from_int, int_from_bracket
from .bracket_probability import bracket_log_probability, bracket_probability
from .bracket_simulation import most_likely_bracket

__all__ = [
    "MAX_BRACKET_INT_EXCLUSIVE",
    "bracket_from_int",
    "int_from_bracket",
    "most_likely_bracket",
    "bracket_log_probability",
    "bracket_probability",
]

File: frontend/src/engine/bracketEngine.js
// bracketEngine.js
//
// Thin façade exporting the public API for the bracket engine.
// Internals are split into smaller modules to keep files maintainable.

export * from "./constants.js";
export { bracketFromInt, intFromBracket } from "./bracketDecode.js";
export { mostLikelyBracket } from "./mostLikely.js";
export { bracketLogProbability, bracketProbability } from "./bracketProbability.js";

import { BRACKETS_PER_PAGE } from "./constants.js";

export function pageFromBracketId(id) {
  // BigInt-safe page index (0-based).
  if (typeof id !== "bigint") throw new TypeError("id must be a BigInt");
  return id / BRACKETS_PER_PAGE;
}

export function firstBracketOnPage(page) {
  if (typeof page !== "bigint") throw new TypeError("page must be a BigInt");
  return page * BRACKETS_PER_PAGE;
}

